tests/test_special_orthogonal.py

```diff
@@ -303,15 +299,14 @@ def test_regularize(self):
                 self.assertAllClose(result, expected)
 
             else:
-                angle = 0.345
+                angle = gs.array(0.345)
```

Comment:
Are all these conversions below necessary? Shouldn't the backends be able to handle number literals the same way primitives like `np.sin` do?

Reply:
If they do, we need to redefine all of these functions for Pytorch and Tensorfow.

Since this is (for now) only for tests, I wonder whether this is worth the trouble ?

Reply:
Right, torch.Tensor(3.4) gives:
```
TypeError: sin(): argument 'input' (position 1) must be Tensor, not float
```

We could create an issue to flag this for later?

Reply:
I guess you mean `torch.sin(1.)`, but yes, that throws an exception, which I find a bit surprising. Still, if we want to mimic the numpy behavior, we should be able to handle such cases without manually having to wrap scalars in the backend's array representation.

Reply:
I am not convinced it is that surprising. torch is made to work with tensors only, to be able to better use GPUs. `numpy.sin(1.)` outputs a float for instance. Since the original point of torch was to do neural networks, the gradient needs to be computed at each step, which requires (or is easier) with the tensor structure I think. And from what a friend told me, Pytorch is also used in other contexts for lazy parallelisation, which is also easier done with tensors (I guess ?)